doc/packages/openfoam-com.section.md

```diff
@@ -0,0 +1,26 @@
+# OpenFOAM (com) {#sec-openfoam-com}
+
+Unfortunately, OpenFOAM binaries assume you execute them via `bash`, after sourcing `$OPENFOAM_DIR/etc/bashrc`.
+To avoid this, binaries from the derivation are wrapped so that they are executed with these requirements.
+
+Additionally, most of its binaries are also able to run in parallel via `mpi`.
+To make invocations easier, each binary is also wrapped to run via `mpi`, such as `simpleFoam-mpi`.
+
+## How to compile and use OpenFOAM extensions {#how-to-compile-and-use-openfoam-com-extensions}
+
+The derivation exposes a function to compile extensions:
+
+```nix
+fooExt = openfoam.mkExtension {
+  name = "foo";
+  src = ./test-ext;
+};
+```
+
+If your extension is an executable, it can be added to `$PATH` as any other executable.
```

Comment:
That's probably more convenient (and predictable, since e.g. Python is a well known example) with a `passthru.withExtensions` helper. Composing extensions in the system environment sounds brittle.

Reply:
I like your suggestion, I didn't think about it. I'll look into how python's `withExtensions` work in order to replicate such behaviour here.